### PR TITLE
Update ZAPD

### PR DESCRIPTION
**Pull request by louist103**
This update adds support for naming various cutscene elements with a config file, improvements to the MM cutscene extraction process, better names for `SkinLimbs`, and updated names for various scene and room structs.

### CODE REVIEW

File: include/z64cutscene.h [hunk 1]
```diff
@@ -631,13 +631,13 @@ typedef struct {
     /* 0x0C */ s16 hudVisibility; 
     /* 0x0E */ u8 endCam;
     /* 0x0F */ u8 letterboxSize;
-} ActorCutscene; // size = 0x10
-// TODO: rename `ActorCutscene` to `CutsceneEntry` once ZAPD uses `CutsceneScriptEntry`
-// typedef CutsceneEntry ActorCutscene;
+} CutsceneEntry; // size = 0x10
+// TODO: rename `CutsceneEntry` to `CutsceneEntry` once ZAPD uses `CutsceneScriptEntry`
+ //typedef CutsceneEntry CutsceneEntry;
```
Comment (hensldm) [inline include/z64cutscene.h:636]: Could you just remove this now?

**Review by hensldm — COMMENTED** [review]
I believe this update should also remove the need for all the other ZAPD compatibility typedefs as well. Would you go through and delete those as well (along with the todo comments)? You can search for `// ZAPD compatibility typedefs` to find them

File: include/z64bgcheck.h [hunk 2]
```diff
@@ -84,7 +84,7 @@ typedef struct {
     /* 0x4 */ Vec3s* bgCamFuncData; // s16 data grouped in threes (ex. Vec3s), is usually of type `BgCamFuncData`
 } BgCamInfo; // size = 0x8
 
-typedef BgCamInfo CamData; // TODO: ZAPD compatibility
+//typedef BgCamInfo CamData; // TODO: ZAPD compatibility
```
Comment (hensldm) [inline include/z64bgcheck.h:87]: Looks like this just got commented out not deleted.

File: include/z64environment.h [hunk 3]
```diff
@@ -211,8 +211,6 @@ typedef struct {
     /* 0x14 */ s16 zFar;
 } EnvLightSettings; // size = 0x16
 
-// ZAPD compatibility typedefs
-// TODO: Remove when ZAPD adds support for them
 typedef CurrentEnvLightSettings LightSettings;
```
Comment (hensldm) [inline include/z64environment.h:214]: Could you add the comments back as well?

**Comment by hensldm** [comment]
Also just found but there are a list of defines in z64cutscene_commands.h prefixed with the comment `// TODO: Fix ZAPD and delete these` that can also now be deleted.

File: include/z64environment.h (hunk 3 above)
Comment (hensldm) [reply include/z64environment.h:214]: Looks like they didn't get added back?

**Review by hensldm — APPROVED** [review]
LGTM, just the ZAPD comments back for
`typedef CurrentEnvLightSettings LightSettings;` please :).

**Comment by louist103** [comment]
> LGTM, just the ZAPD comments back for `typedef CurrentEnvLightSettings LightSettings;` please :).

It won't build without that for me.

**Comment by hensldm** [comment]
> > LGTM, just the ZAPD comments back for `typedef CurrentEnvLightSettings LightSettings;` please :).
> 
> It won't build without that for me.

Yeah don't remove the typedef, just add the ZAPD compatibility and todo comments back

**Review by engineer124 — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**